src/main.zig

```diff
@@ -924,6 +928,15 @@ fn buildOutputType(
                         search_strategy = .paths_first;
                     } else if (mem.eql(u8, arg, "-search_dylibs_first")) {
                         search_strategy = .dylibs_first;
+                    } else if (mem.eql(u8, arg, "-headerpad")) {
+                        const next_arg = args_iter.next() orelse {
+                            fatal("expected parameter after {s}", .{arg});
+                        };
+                        headerpad_size = std.fmt.parseUnsigned(u32, eatIntPrefix(next_arg, 16), 16) catch |err| {
+                            fatal("unable to parser '{s}': {s}", .{ arg, @errorName(err) });
+                        };
+                    } else if (mem.eql(u8, arg, "-headerpad_max_install_names")) {
+                        headerpad_max_install_names = true;
```

Comment:
nit: do we want to allow naked `-headerpad` and `-headerpad_max_install_names` to `zig cc`, or should we require them to be passed via `-Wl,<...>` alone?

I have done this mistake myself a few times of adding a linker-only option here (will submit PRs reverting that soon); but we should be quite upfront about adding them here.

Reply:
Turns out none of such 'bad" code is landed yet; I've updated the PRs-in-flight to the correct behavior.

Reply:
This PR added support for both. `build-*` accepts directly `-headerpad` and `-headerpad_max_install_names` https://github.com/ziglang/zig/blob/08459ff1c21d546c55e2ae954126e121ee88972e/src/main.zig#L931, and `cc` and `c++` accept `-Wl,-headerpad=<size>` and `-Wl,-headerpad_max_install_names` https://github.com/ziglang/zig/blob/08459ff1c21d546c55e2ae954126e121ee88972e/src/main.zig#L1692 :^)